### DISCUSSION TITLE
fileUpload , cant upload large PDF file with 33.5 MB size

### DISCUSSION THREAD

**Discussion by Yazan-Abuawwad** [primefaces/community#1820] [PrimeFaces] [answered]
fileUpload , cant upload large PDF file  with  33.5 MB size   
primface 13 , java 17 ,  tomcat , no error in the console and all configuration in web.xml like doc server API
the max limit success with me is 1 MB and I don't know why

<p:fileUpload listener="#{patientMB.handleFileUpload}" mode="advanced"
update="messages" auto="true" sizeLimit="999999999" allowTypes="/(.|/)(pdf)$/"/>
                              
                              
![image](https://github.com/primefaces/community/assets/88251354/7993e06e-01b9-49f9-bdbf-0b8c9b256b5d)

**Comment by Rapster** [comment]
Remove binding, don't think that's what you want to do... And try without maxChunkSize also to see what happens (FYI, 34 MB is not large)

**Reply by Yazan-Abuawwad** [reply]
I know binding it doesn't benefit I was trying  anything 
this is what is on my page 
 <p:fileUpload listener="#{patientMB.handleFileUpload}" mode="advanced"
                                      update="messages" auto="true" sizeLimit="999999999" allowTypes="/(\.|\/)(pdf)$/"/>

**Reply by Rapster** [reply]
Make sure it's enclosed in a `<h:form enctype="multipart/form-data">` or provide a reproducer using primefaces-test

**Reply by Yazan-Abuawwad** [reply]
with or without <h:form not working for me 
maybe i need properties in yml file bc i used joinface ?

**Reply by melloware** [reply, marked as answer]
If you are using JoinFaces you may have to increase Tomcat/Jetty/Undertow depending on what Servlet engine you are using with Joinfaces to allow larger than 10MB transfers.   For example

```yaml
spring:
  servlet:
    multipart:
      max-file-size: 100MB
      max-request-size: 100MB
```
